src/app/component/CategoryProductList/CategoryProductList.component.js

```diff
@@ -22,154 +24,236 @@ import './CategoryProductList.style';
-              elem="NoProducts"
-            >
-                No products found
-            </div>
-        );
+        const keys = Object.keys(pages);
+
+        const maxPage = Math.max(...keys);
+        const minPage = Math.min(...keys);
+
+        const loadedPagesCount = keys.length;
+
+        return { maxPage, minPage, loadedPagesCount };
     }
 
-    renderProducts() {
-        const { items, customFilters } = this.props;
+    /**
+     * Observes new pages and unobserves obsolete
+     */
+    updateObserver() {
+        const currentNodes = Object.values(this.nodes);
+
+        if (!this.observer || currentNodes.length <= 0) return;
 
-        if (items.length === 0) return this.renderNoProducts();
+        currentNodes.forEach((node) => {
+            if (node && !this.observedNodes.includes(node)) {
+                this.observer.observe(node);
+                this.observedNodes.push(node);
+            }
+        });
+
+        this.observedNodes = this.observedNodes.reduce((acc, node) => {
+            if (!currentNodes.includes(node)) {
+                this.observer.unobserve(node);
+            } else {
+                acc.push(node);
+            }
 
-        return items.map(product => (
-            <ProductCard
-              product={ product }
-              key={ product.id }
-              customFilters={ customFilters }
-              arePlaceholdersShown
-            />
-        ));
+            return acc;
+        }, []);
     }
 
     /**
-     * render placeholders beneath the product list
+     * Loads previous or next page
+     * @param {Boolean} next
      */
-    renderPlaceholder(showLoadMore, isLoading) {
-        const renderPlaceholders = () => (
-            <>
-                <ProductCard product={ {} } />
-                <ProductCard product={ {} } />
-                <ProductCard product={ {} } />
-            </>
-        );
+    loadPage(next = true) {
+        const { pagesCount } = this.state;
+        const { loadPage, totalPages, isLoading } = this.props;
+        const { minPage, maxPage, loadedPagesCount } = this.getPagesBounds();
 
-        if (isLoading) {
-            return renderPlaceholders();
+        const isUpdateable = totalPages > 0 && pagesCount === loadedPagesCount;
+        const shouldUpdateList = next ? maxPage < totalPages : minPage > 1;
+
+        if (isUpdateable && shouldUpdateList && !isLoading) {
+            const modifier = next ? maxPage + 1 : minPage - 1;
+
+            this.setState({ pagesCount: pagesCount + 1 });
+            loadPage(modifier);
         }
+    }
 
-        if (showLoadMore) {
-            return (
+    renderLoadButton(next = false, hidden = false) {
+        const loadPage = this.loadPage.bind(null, next);
+        return (
+            <div
+              block="CategoryProductList"
+              elem="LoadButton"
+              mods={ { hidden } }
+              role="button"
+              tabIndex="-1"
```

Comment:
Not sure if this is good for SEO, could you please check Googles implementation?

Reply:
I see people shouting `don't use tabIndex greater that zero`, but there is no complains about index -1. I am referring to https://dequeuniversity.com/rules/axe/3.2/tabindex and https://developers.google.com/web/tools/lighthouse/audits/tabindex I will set it to zero, just the question of necessity.